src/uint/div.rs

```diff
@@ -17,35 +17,29 @@ impl<const LIMBS: usize> UInt<LIMBS> {
     /// When used with a fixed `rhs`, this function is constant-time with respect
     /// to `self`.
     pub(crate) const fn ct_div_rem(&self, rhs: &Self) -> (Self, Self, u8) {
-        let mut bd = self.bits().saturating_sub(rhs.bits());
+        let mb = rhs.bits();
+        let mut bd = (LIMBS * Limb::BIT_SIZE) - mb;
         let mut rem = *self;
         let mut quo = Self::ZERO;
-
         let mut c = rhs.shl_vartime(bd);
-        let mut e = Self::ONE.shl_vartime(bd);
 
         loop {
-            let mut r: Self = rem.wrapping_sub(&c);
-            let d = -(((1 - (r.limbs[LIMBS - 1].0 >> HI_BIT)) & 1) as SignedWord);
-            let d = d as Word;
-            rem = Self::ct_select(rem, r, d);
-            r = quo;
-            r = r.wrapping_add(&e);
-            quo = Self::ct_select(quo, r, d);
+            let (mut r, borrow) = rem.sbb(&c, Limb::ZERO);
+            rem = Self::ct_select(r, rem, borrow.0);
+            r = quo.bitor(&Self::ONE);
+            quo = Self::ct_select(r, quo, borrow.0);
             if bd == 0 {
                 break;
             }
             bd -= 1;
             c = c.shr_vartime(1);
-            e = e.shr_vartime(1);
+            quo = quo.shl_vartime(1);
+        }
+
+        let is_some = if mb == 0 { 0 } else { 1 };
```

Comment:
I believe this is sufficient since there are no constant-time guarantees around `rhs`.

Reply:
Might be good to leave a comment to that effect

Reply:
I threw in a `ct_select` since it doesn't make a significant difference to the runtime.

Reply:
Can you add your test case that triggered this PR?

Reply:
Want to make sure it’s actually fixed

Reply:
Just saw this comment.

Perhaps it would make sense to use something like [`sidefuzz`](https://github.com/phayes/sidefuzz) to test constant-time behvaior?

Reply:
My test was a criterion benchmark, I suspected it from reading the code.

```rs
#[macro_use]
extern crate criterion;
use criterion::{black_box, Criterion};
fn criterion_benchmark(c: &mut Criterion) {
    use crypto_bigint::U256;
    let u1 = U256::from_be_hex("0123456780123456780123456780123456780123456780123456780123456780");
    let u2 = U256::from_be_hex("0000000000000000000000000000000000000000000000000000000000000012");
    let md = U256::from_be_hex("0000000000000000000000000000000000000000000000000000000012345678");
    c.bench_function("reduce large", move |b| {
        b.iter(|| black_box(u1).reduce(&md))
    });
    c.bench_function("reduce small", move |b| {
        b.iter(|| black_box(u2).reduce(&md))
    });
}
criterion_group!(benches, criterion_benchmark);
criterion_main!(benches);
```

Old timing:

```
reduce large            time:   [948.59 ns 949.00 ns 949.46 ns]                           
Found 8 outliers among 100 measurements (8.00%)
  6 (6.00%) high mild
  2 (2.00%) high severe

reduce small            time:   [15.520 ns 15.535 ns 15.549 ns]                           
Found 3 outliers among 100 measurements (3.00%)
  1 (1.00%) low mild
  2 (2.00%) high severe
```

And patched:

```
reduce large            time:   [955.12 ns 955.84 ns 956.81 ns]                          
Found 10 outliers among 100 measurements (10.00%)
  3 (3.00%) high mild
  7 (7.00%) high severe

reduce small            time:   [955.58 ns 956.55 ns 957.99 ns]                          
Found 11 outliers among 100 measurements (11.00%)
  4 (4.00%) high mild
  7 (7.00%) high severe
```

Reply:
Also if somebody feels like implementing a limb-based method this looks interesting: https://www.bearssl.org/bigint.html#modred